compiler/daml-extension/BUILD.bazel

```diff
@@ -66,10 +66,11 @@ genrule(
             cp ../$$f $$dir/$$file
         done
         cd ..
-        tar c node_modules \\
-          --owner=0 --group=0 --numeric-owner --mtime=2000-01-01\\ 00:00Z --sort=name \\
-          | gzip -n > $@
+        # Avoid file path too long errors on Windows of the form
+        #   .../tar_dev_env/usr/bin/tar: node_modules/.cache/terser-webpack-plugin/...: file name is too long (cannot be split); not dumped
```

Comment:
This is a consequence of `--format=ustar`. Looking at IEEE Std 1003.1-1988 the filename is limited to 256 characters split into two components separated by `/`: a prefix of at most 155 characters and the filename of at most 100 characters. In this case the filename `97e9576e421fc74fc3b3eed0f6f0ab3be937d0f110bacca2151119236a1a6fc2a5262b044ec008be7300e3489eb49631d1f0572d08ca99384f2f9182ab57` exceeds the 100 character limit.

If need longer file names at some point then [reproducible-builds.org](https://reproducible-builds.org/docs/archives/) recommends

> To avoid this, either unset POSIXLY_CORRECT (only works with tar>1.32) or add to the tar call --pax-option=exthdr.name=%d/PaxHeaders/%f,delete=atime,delete=ctime or --format=gnu (both only available in GNU tar) or use --format=ustar if the limitations in that format are not a problem.

to avoid non-deterministic PAX headers. However, it places some restrictions on the tar version.

Reply:
Makes sense, thanks for the explanation!